From f0d5f91d82de8b49e1f345505214960289fd85d8 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Mon, 25 Jul 2016 13:25:30 +0200
Subject: [PATCH 1/8] zynqmp: Boot from SD card if SPI is not supported

QSPI support for all Zynq devices in SPL is still broken. Fall through to
SD card booting to at least speed up the boot a bit.
---
 arch/arm/cpu/armv8/zynqmp/spl.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/arch/arm/cpu/armv8/zynqmp/spl.c b/arch/arm/cpu/armv8/zynqmp/spl.c
index bc7313a..4a36731 100644
--- a/arch/arm/cpu/armv8/zynqmp/spl.c
+++ b/arch/arm/cpu/armv8/zynqmp/spl.c
@@ -82,6 +82,12 @@ u32 spl_boot_device(void)
 	switch (bootmode) {
 	case JTAG_MODE:
 		return BOOT_DEVICE_RAM;
+	case QSPI_MODE_24BIT:
+	case QSPI_MODE_32BIT:
+	/* When SPI support isn't enabled, fall through to SD/MMC boot */
+#ifdef CONFIG_SPL_SPI_SUPPORT
+		return BOOT_DEVICE_SPI;
+#endif
 #ifdef CONFIG_SPL_MMC_SUPPORT
 	case SD_MODE1:
 	case SD1_LSHFT_MODE: /* not working on silicon v1 */
@@ -102,11 +108,6 @@ u32 spl_boot_device(void)
 	case SW_SATA_MODE:
 		return BOOT_DEVICE_SATA;
 #endif
-#ifdef CONFIG_SPL_SPI_SUPPORT
-	case QSPI_MODE_24BIT:
-	case QSPI_MODE_32BIT:
-		return BOOT_DEVICE_SPI;
-#endif
 	default:
 		printf("Invalid Boot Mode:0x%x\n", bootmode);
 		break;
-- 
1.9.1

